### PR TITLE
BUG: fix colorbar minor ticks drawing for symlog norms

### PR DESCRIPTION
**Pull request by neutrinoceros**
## PR Summary

This is the patch I suggested in #3535
As far as I can tell the bug is very likely to be in Matplotlib itself, though I wasn't able to replicate it in isolation for now.
I'm opening this a a draft while I attempt to show that this buggy behaviour can be reproduced without yt, in which case I'll report upstream and link the report in a comment.

### CODE REVIEW

**Review by matthewturk — APPROVED** [review]
Good catch.  Assuming you move this to non-draft, LGTM.

**Comment by neutrinoceros** [comment]
So I'm actually going to change this to a slightly less aggressive form. I still don't have a clear matplotlib-isolated example but I'm now certain that the bug is on their side and was introduced in matplotlib 3.2.0
Here's the results of your script from #3535 using two different combinations of versions:
 
### yt 4.0.1 + matplotlib 3.2.0
symlog

### yt 4.0.1 + matplotlib 3.1.3 (latest version before 3.2.0)
symlog

**Comment by neutrinoceros** [comment]
Making a matplolitb-only example of this bug turns out to be much more challenging than I would have guessed, so I won't be able to report it properly for now, but I can at least avoid disable minor ticks drawing with Matplotlib < 3.2, so I made the skipping conditional. Hopefully the comment is clear enough for the next person to inquire this in the future.
There's exactly one failing test with my change, and I think we can agree that the new version is preferable to the current baseline.
See
https://tests.yt-project.org/job/yt_py38_git/3861/testReport/junit/yt.geometry.coordinates.tests/test_cylindrical_coordinates/GenericImage_AMRGridData_all_create_image_None_None/

@matthewturk, do I have your approval to bump this baseline ?

**Comment by neutrinoceros** [comment]
@chummels you might want to review this too

**Comment by neutrinoceros** [comment]
thanks to a discussion with matplotlib maintainers https://github.com/matplotlib/matplotlib/issues/21258#issuecomment-932699643, I got an alternative solution for the upcoming matplotlib 3.5 that almost works.
symlog
So now there's some useless whitespace at the top of the colorbar, but I believe this is a bug in yt, so now let's try to fix this as well.

**Comment by neutrinoceros** [comment]
fixed that last issue in https://github.com/yt-project/yt/pull/3536/commits/e317c03e54b67137d9aa01b9acf7aa565078777d
The commit has no effect with current versions of matplotlib but fixes the issue with the upcoming matplotlib 3.5 noted in my previous comment

**Review by chummels — APPROVED** [review]
Looks good.  Thanks for digging into this and finding a workable solution!

**Pull request merged**
